server/publicapi/items/service.py

```diff
@@ -73,27 +73,27 @@ def get(self, req, lookup):
             req = ParsedRequest()
 
         allowed_params = (
-            'q', 'start_date', 'end_date',
+            'q', 'start_date', 'end_date', 'filter',
             'include_fields', 'exclude_fields'
         )
         self._check_for_unknown_params(req, whitelist=allowed_params)
 
-        # set the "q" filter
         request_params = req.args or {}
-        query_filter = {}
+        req.args = {}
 
+        # set the search query
         if 'q' in request_params:
-            # TODO: add validation for the "q" parameter when we define its
-            # format and implement the corresponding actions
-            query_filter = json.loads(request_params['q'])
+            req.args['q'] = request_params['q']
```

Comment:
So it's probably the backend storage that will throw an error in case of a malformed value? And that would most likely result in some kind of an internal server error? I think we should make sure that InvalidParameterValueError is reported back in such cases.

Can you check if this is currently the case?